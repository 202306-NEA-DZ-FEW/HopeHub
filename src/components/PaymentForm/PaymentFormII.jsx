import { useTranslation } from "next-i18next";
import React, { useState } from "react";
import Select from "react-select";
import countryList from "react-select-country-list";

const PaymentFormII = () => {
    const { t } = useTranslation("common");

    const [isMasterCard, setIsMasterCard] = useState(false);
    const [cardType, setCardType] = useState("Visa");
    const [cardNumber, setCardNumber] = useState("");
    const [expiryDate, setExpiryDate] = useState("");
    const [cvv, setCvv] = useState("");
    const [nameOnCard, setNameOnCard] = useState("");
    const [zipCode, setZipCode] = useState("");
    const [city, setCity] = useState("");
    const [address, setAddress] = useState("");
    const [selectedCountry, setSelectedCountry] = useState(null);

    const handleCountryChange = (selectedOption) => {
        setSelectedCountry(selectedOption);
    };

    const handleCardChange = () => {
        setIsMasterCard(!isMasterCard);
        const card = isMasterCard ? "masterCard" : "Visa";
        setCardType(card);
    };

    const handleExpiryDateChange = (e) => {
        const input = e.target.value;

        // Allow only digits and "/" character
        if (/^\d{0,2}(\/\d{0,2})?$/.test(input)) {
            setExpiryDate(input);
        }
    };

    const handleSubmit = (e) => {
        e.preventDefault();

        if (expiryDate.length !== 5) {
            console.log("Invalid expiry date format");
            return;
        }
        if (!selectedCountry) {
            console.log("Please select a country");
            return;
        }
        // Handle the form submission here
    };

    return (
        <div className='w-full'>
            {/* Left Column */}

            <form
                className='w-full flex flex-col gap-5 md:flex-row justify-start items-center lg:gap-32'
                onSubmit={handleSubmit}
            >
                <div className='w-full flex flex-col justify-start'>
                    <label htmlFor='cardType' className='text-lg'>
                        {t("Supported Card types")}
                    </label>
                    <div className='flex flex-row items-center justify-start '>
                        <button
                            onClick={handleCardChange}
                            className={`block w-1/2 p-2 border border-Accent rounded rounded-tr-none rounded-br-none ${
                                isMasterCard
                                    ? "bg-blue-100 text-Accent"
                                    : "text-blue-400"
                            }`}
                        >
                            {t("Visa")}
                        </button>

                        <button
                            onClick={handleCardChange}
                            className={`block w-1/2 p-2 border border-Accent rounded rounded-tl-none rounded-bl-none ${
                                isMasterCard
                                    ? "text-Accent "
                                    : "bg-blue-100 text-Accent"
                            }`}
                        >
                            {t("MasterCard")}
                        </button>
                    </div>
                    <div className='mt-4'>
                        <label htmlFor='cardNumber' className='text-lg'>
                            {t("Card Number")}
                        </label>
                        <input
                            type='number'
                            id='cardNumber'
                            value={cardNumber}
                            onChange={(e) => {
                                if (e.target.value.length <= 14) {
                                    setCardNumber(e.target.value);
                                }
                            }}
                            className='block w-full p-2  bg-white rounded border border-Accent focus:border-red-500 outline-none'
                            maxLength='14'
                            pattern='[0-9]{14}'
                            title='Card number must be 14 digits'
                            placeholder='**************'
                        />
                    </div>
                    <div className='flex flex-wrap mt-4'>
                        <div className='w-1/2 pr-2'>
                            <label
                                htmlFor='expiryDate mb-2'
                                className='text-lg'
                            >
                                {t("Expiry Date")}
                            </label>
                            <input
                                type='text'
                                id='expiryDate'
                                value={expiryDate}
                                onChange={handleExpiryDateChange}
                                className='block w-full p-2 bg-white rounded border border-Accent focus:border-red-500 outline-none'
                                placeholder={t("MM/YY")}
                            />
                        </div>
                        <div className='w-1/2 pl-2'>
                            <label htmlFor='cvv' className='text-lg'>
                                {t("CVV Code")}
                            </label>
                            <input
                                type='number'
                                id='cvv'
                                value={cvv}
                                onChange={(e) => setCvv(e.target.value)}
                                className='block w-full p-2 bg-white rounded border border-Accent focus:border-red-500 outline-none'
                                maxLength={3}
                                pattern='[0-9]{3}'
                                title='CVV number must be 3 digits'
                                placeholder='***'
                            />
                        </div>
                    </div>
                    <div className='mt-4'>
                        <label htmlFor='nameOnCard' className='text-lg'>
                            {t("Name on Card")}
                        </label>
                        <input
                            type='text'
                            id='nameOnCard'
                            value={nameOnCard}
                            onChange={(e) => setNameOnCard(e.target.value)}
                            className='block w-full p-2 bg-white rounded border border-Accent focus:border-red-500 outline-none'
                            placeholder={t("Name On Card")}
                        />
                    </div>
                </div>

                {/* Right Column */}
                <div className='w-full flex flex-col justify-start'>
                    <div className=''>
                        <label htmlFor='country' className='text-lg'>
                            {t("Country")}
                        </label>
                        <Select
                            id='country'
                            value={selectedCountry}
                            onChange={handleCountryChange}
                            options={countryList().getData()}
                            // className="block w-full p-2 bg-white rounded border-[1px] border-Accent focus-border-red-500"
                            placeholder={t("Your Country Name")}
                        />
                    </div>
                    <div className='mt-4'>
                        <label htmlFor='zipCode' className='text-lg'>
                            {t("Zip Code")}
                        </label>
                        <input
                            type='text'
                            id='zipCode'
                            value={zipCode}
                            onChange={(e) => setZipCode(e.target.value)}
                            className='block w-full p-2 bg-white rounded border border-Accent focus:border-red-500 outline-none'
                            placeholder='*********'
                        />
                    </div>

                    <div className='mt-4'>
                        <label htmlFor='city' className='text-lg'>
                            {t("City")}
                        </label>
                        <input
                            type='text'
                            id='city'
                            value={city}
                            onChange={(e) => setCity(e.target.value)}
                            className='block w-full p-2 bg-white rounded border border-Accent focus:border-red-500 outline-none'
                            placeholder={t("Your City Name")}
                        />
                    </div>
                    <div className='mt-4'>
                        <label htmlFor='address' className='text-lg'>
                            {t("Address")}
                        </label>
                        <input
                            type='text'
                            id='address'
                            value={address}
                            onChange={(e) => setAddress(e.target.value)}
                            className='block w-full p-2 bg-white rounded border border-Accent focus:border-red-500 outline-none'
                            placeholder={t("Your Address")}
                        />
                    </div>
                </div>
            </form>
            <div>
                <button
                    type='submit'
                    className='w-32 p-2 bg-Accent text-NeutralBlack font-bold rounded mt-8'
                >
                    {t("Add Card")}
                </button>
            </div>
        </div>
    );
};

export default PaymentFormII;
