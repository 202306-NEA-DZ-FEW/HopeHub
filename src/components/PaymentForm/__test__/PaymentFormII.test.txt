import renderer from "react-test-renderer";

import PaymentFormII from "../PaymentFormII";

it("renders correctly", () => {
    const tree = renderer.create(<PaymentFormII />).toJSON();
    expect(tree).toMatchSnapshot();
});
