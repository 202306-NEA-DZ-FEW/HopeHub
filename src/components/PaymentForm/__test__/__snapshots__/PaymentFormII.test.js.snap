// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="w-full"
>
  <form
    className="w-full flex flex-col gap-5 md:flex-row justify-start items-center lg:gap-32"
    onSubmit={[Function]}
  >
    <div
      className="w-full flex flex-col justify-start"
    >
      <label
        className="text-lg"
        htmlFor="cardType"
      >
        Supported Card types
      </label>
      <div
        className="flex flex-row items-center justify-start "
      >
        <button
          className="block w-1/2 p-2 border border-Accent rounded rounded-tr-none rounded-br-none text-blue-400"
          onClick={[Function]}
        >
          Visa
        </button>
        <button
          className="block w-1/2 p-2 border border-Accent rounded rounded-tl-none rounded-bl-none bg-blue-100 text-Accent"
          onClick={[Function]}
        >
          MasterCard
        </button>
      </div>
      <div
        className="mt-4"
      >
        <label
          className="text-lg"
          htmlFor="cardNumber"
        >
          Card Number
        </label>
        <input
          className="block w-full p-2  bg-white rounded border border-Accent focus:border-red-500 outline-none"
          id="cardNumber"
          maxLength="14"
          onChange={[Function]}
          pattern="[0-9]{14}"
          placeholder="**************"
          title="Card number must be 14 digits"
          type="number"
          value=""
        />
      </div>
      <div
        className="flex flex-wrap mt-4"
      >
        <div
          className="w-1/2 pr-2"
        >
          <label
            className="text-lg"
            htmlFor="expiryDate mb-2"
          >
            Expiry Date
          </label>
          <input
            className="block w-full p-2 bg-white rounded border border-Accent focus:border-red-500 outline-none"
            id="expiryDate"
            onChange={[Function]}
            placeholder="MM/YY"
            type="text"
            value=""
          />
        </div>
        <div
          className="w-1/2 pl-2"
        >
          <label
            className="text-lg"
            htmlFor="cvv"
          >
            CVV Code
          </label>
          <input
            className="block w-full p-2 bg-white rounded border border-Accent focus:border-red-500 outline-none"
            id="cvv"
            maxLength={3}
            onChange={[Function]}
            pattern="[0-9]{3}"
            placeholder="***"
            title="CVV number must be 3 digits"
            type="number"
            value=""
          />
        </div>
      </div>
      <div
        className="mt-4"
      >
        <label
          className="text-lg"
          htmlFor="nameOnCard"
        >
          Name on Card
        </label>
        <input
          className="block w-full p-2 bg-white rounded border border-Accent focus:border-red-500 outline-none"
          id="nameOnCard"
          onChange={[Function]}
          placeholder="Name On Card"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="w-full flex flex-col justify-start"
    >
      <div
        className=""
      >
        <label
          className="text-lg"
          htmlFor="country"
        >
          Country
        </label>
        <div
          className=" css-b62m3t-container"
          id="country"
          onKeyDown={[Function]}
        >
          <span
            className="css-1f43avz-a11yText-A11yText"
            id="react-select-2-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            className="css-1f43avz-a11yText-A11yText"
            role="log"
          />
          <div
            className=" css-13cymwt-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className=" css-1fdsijx-ValueContainer"
            >
              <div
                className=" css-1jqq78o-placeholder"
                id="react-select-2-placeholder"
              >
                Your Country Name
              </div>
              <div
                className=" css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-activedescendant=""
                  aria-autocomplete="list"
                  aria-describedby="react-select-2-placeholder"
                  aria-expanded={false}
                  aria-haspopup={true}
                  autoCapitalize="none"
                  autoComplete="off"
                  autoCorrect="off"
                  className=""
                  disabled={false}
                  id="react-select-2-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="combobox"
                  spellCheck="false"
                  style={
                    Object {
                      "background": 0,
                      "border": 0,
                      "color": "inherit",
                      "font": "inherit",
                      "gridArea": "1 / 2",
                      "label": "input",
                      "margin": 0,
                      "minWidth": "2px",
                      "opacity": 1,
                      "outline": 0,
                      "padding": 0,
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className=" css-1u9des2-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className=" css-1xc3v61-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-tj5bde-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="mt-4"
      >
        <label
          className="text-lg"
          htmlFor="zipCode"
        >
          Zip Code
        </label>
        <input
          className="block w-full p-2 bg-white rounded border border-Accent focus:border-red-500 outline-none"
          id="zipCode"
          onChange={[Function]}
          placeholder="*********"
          type="text"
          value=""
        />
      </div>
      <div
        className="mt-4"
      >
        <label
          className="text-lg"
          htmlFor="city"
        >
          City
        </label>
        <input
          className="block w-full p-2 bg-white rounded border border-Accent focus:border-red-500 outline-none"
          id="city"
          onChange={[Function]}
          placeholder="Your City Name"
          type="text"
          value=""
        />
      </div>
      <div
        className="mt-4"
      >
        <label
          className="text-lg"
          htmlFor="address"
        >
          Address
        </label>
        <input
          className="block w-full p-2 bg-white rounded border border-Accent focus:border-red-500 outline-none"
          id="address"
          onChange={[Function]}
          placeholder="Your Address"
          type="text"
          value=""
        />
      </div>
    </div>
  </form>
  <div>
    <button
      className="w-32 p-2 bg-Accent text-NeutralBlack font-bold rounded mt-8"
      type="submit"
    >
      Add Card
    </button>
  </div>
</div>
`;
